src/App.tsx

```diff
@@ -1,54 +1,128 @@
+    [],
+  );
+
+  const handleQuaryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuary(event.target.value);
+    setTouched(false);
+  };
+
+  function getPersonBySlug(id: string) {
+    const foundPerson = filteredPeople.find(
+      person => person.slug === id,
+    ) || null;
+
+    return foundPerson;
+  }
+
+  const handleOnMouseDown = (slug: string) => {
+    const foundPerson = getPersonBySlug(slug);
+
+    setUserId(slug);
+    setFindedPerson(foundPerson);
+    setQuery(foundPerson?.name || '');
+    setAppliedQuary(foundPerson?.name || '');
+  };
 
   return (
     <main className="section">
-      <h1 className="title">
-        {`${name} (${born} = ${died})`}
-      </h1>
+      {userId
+        ? (
+          <h1 className="title">
+            {`${findedPerson?.name} (${findedPerson?.born} - ${findedPerson?.died})`}
+          </h1>
+        )
+        : (
+          <p>No selected person</p>
+        )}
 
       <div className="dropdown is-active">
         <div className="dropdown-trigger">
           <input
             type="text"
+            value={query}
             placeholder="Enter a part of the name"
             className="input"
+            onChange={handleQuaryChange}
+            onFocus={() => setTouched(true)}
+            onBlur={() => setTouched(false)}
           />
         </div>
 
         <div className="dropdown-menu" role="menu">
           <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
+            {touched && (
+              filteredPeople.map(person => (
+                <div className="dropdown-item" key={person.slug}>
+                  <button
+                    type="button"
+                    className="has-text-link"
+                    onMouseDown={() => handleOnMouseDown(person.slug)}
+                  >
+                    {person.name}
+                  </button>
+                </div>
+              ))
+            )}
 
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
+            {filteredPeople.length === 0 && (<p>No matching suggestions</p>)}
```

Comment:
```suggestion
            {!filteredPeople.length && (<p>No matching suggestions</p>)}
```

let's simplify this line